### PR TITLE
Implementing ShouldBe with Case choice for IEnumerable<string>

### PR DESCRIPTION
**Pull request by JulienFiaffe**
Hello @JakeGinnivan,
This is a Pull Request for issue #244.

I have added the 3 following extensions, as specified in the issue.

```csharp
static void ShouldBe(this IEnumerable<string> actual, IEnumerable<string> expected, Case caseSensitivity);
static void ShouldBe(this IEnumerable<string> actual, IEnumerable<string> expected, Case caseSensitivity, string customMessage);
static void ShouldBe(this IEnumerable<string> actual, IEnumerable<string> expected, Case caseSensitivity, [InstantHandle] Func<string> customMessage);
```

I also have added test scenarios for these new methods. I chose to add a new validation method in `Is` named `EnumerableStringEqualWithCaseSensitivity`. Let me know if it follows the project standards.

Do not hesitate to ask me anything if needed.

### CODE REVIEW

File: src/Shouldly.Tests/ShouldBe/EnumerableType/EnumerableOfStringIsInsensitiveScenario.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        protected override void ShouldThrowAWobbly()
+        {
+            new[] { "foo" }.ShouldBe(new[] { "different" }, Case.Insensitive, () => "Some additional context");
+        }
+
+        protected override string ChuckedAWobblyErrorMessage
+        {
+            get
+            {
+                return "new[] { \"foo\" } should be [\"different\"] but was [\"foo\"] difference [*\"foo\"*]" +
```
Comment (JakeGinnivan) [inline src/Shouldly.Tests/ShouldBe/EnumerableType/EnumerableOfStringIsInsensitiveScenario.cs]: I think the error should show if it is a case sensitive or insensitive comparison.

File: src/Shouldly/Internals/Is.cs [hunk 2]
```diff
@@ -236,6 +236,26 @@ public static bool StringEqualWithCaseSensitivity(string actual, string expected
...
+
+            for (; ; )
+            {
+                var expectedHasData = expectedEnum.MoveNext();
+                var actualHasData = actualEnum.MoveNext();
+
+                if (!expectedHasData && !actualHasData)
+                    return true;
+
+                if (!StringEqualWithCaseSensitivity(actualEnum.Current, expectedEnum.Current, caseSensitivity))
```
Comment (JakeGinnivan) [inline src/Shouldly/Internals/Is.cs]: This will throw if the lists are different lengths

**Comment by JakeGinnivan** [comment]
Looks pretty good. I have left this for ages so if you don't have time to finish this off let me know and I can make the required changes.

**Comment by JakeGinnivan** [comment]
@JulienFiaffe are you able to look at the comments, otherwise I can do it. Let me know either way

**Comment by JulienFiaffe** [comment]
Hi, thank you for the answer, I am abroad, I will have a look on Sunday when I'll get back.

**Comment by JulienFiaffe** [comment]
@JakeGinnivan, than you for the answers. I made the change regarding the lists lengths. On the other side I think I will let you update the error message for the case sensitive/insensitive comparison, I do not know this part enough to make an update.
Let me know if you need information.

**Comment by JoeMighty** [comment]
I'm happy to finish this one off, I've done some case sensitive/insensitive stuff before so know what to do.

**Comment by JakeGinnivan** [comment]
Thanks @JoeMighty!

**Comment by JoeMighty** [comment]
I've just committed the error message changes to @JulienFiaffe's fork (I presume he'll have to merge the PR before it shows in this PR?).

Having never created a pull request on someone else's pull request before I wasn't sure on the git workflow.

**Comment by JulienFiaffe** [comment]
@JoeMighty Thank you for the help.
I made the merge to the branch for the pull request.

**Pull request merged**